### PR TITLE
View definition modal for make url endpoint.

### LINKED ISSUE #13: Export data cube definition from UI
New feature to export current data cube definition from UI as JSON (format compatible with mkurl payload). Right now it is really hard to figure out payload for mkurl endpoint. With this feature the existing visualisation would be used as a prototype for mkurl.

UI design proposal: 
New action "Display view definition" under "Share link" menu, cube definition should be presented in the new modal window (the last element in the menu). 
Change "View raw data" into "Display raw data".

Nice to have:
JSON syntax highlighting.

### PR DESCRIPTION
**Pull request by rafalnowak**
Fixes #13

### CODE REVIEW

File: src/client/components/hiluk-menu/hiluk-menu.tsx [hunk 1]
```diff
@@ -92,61 +98,60 @@ export class HilukMenu extends React.Component<HilukMenuProps, HilukMenuState> {
...
-    var shareOptions: JSX.Element[] = [];
+    const shareOptions: JSX.Element[] = [];
 
     if (addEssenceToCollection) {
       shareOptions.push(<li
         key="add-to-collection"
         className="add-to-collection"
-        onClick={addEssenceToCollection}
-      >{STRINGS.addToCollection}</li>);
+        onClick={addEssenceToCollection}>{STRINGS.addToCollection}</li>);
```
Comment (piter75) [inline src/client/components/hiluk-menu/hiluk-menu.tsx]: why joining these lines? isn't the previous version more readable?

File: src/client/config/constants.ts [hunk 2]
```diff
@@ -137,7 +137,11 @@ export const STRINGS: any = {
   timezone: 'Timezone',
   undo: 'Click here to undo',
   updateTimezone: 'Update Timezone',
-  viewRawData: 'View raw data',
+  displayRawData: 'Display raw data',
+  displayViewDefinition: 'Display view definition',
+  viewDefinition: 'View definition',
+  viewDefinitionSubtitle: 'View definition for mkurl',
+  mkurlDomainPlaceholder: 'CHANGE ME',
```
Comment (piter75) [inline src/client/config/constants.ts]: why not keeping an alphabetical order?

File: src/client/modals/view-definition-modal/view-definition-modal.tsx [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+
+import * as React from 'react';
+import {Essence, EssenceJS} from '../../../common/models';
+
+import {Fn, makeTitle} from '../../../common/utils';
+import {STRINGS} from '../../config/constants';
+
+import {Button, Modal} from '../../components';
+
+export interface ViewDefinitionModalProps extends React.Props<any> {
```
Comment (piter75) [inline src/client/modals/view-definition-modal/view-definition-modal.tsx]: React.Props is deprecated, you don't have to extend it

File: src/client/modals/view-definition-modal/view-definition-modal.tsx [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+  }
+
+  public printAsJson(): string {
+    return JSON.stringify(this, null, 2);
+  }
+}
+
+export class ViewDefinitionModal extends React.Component<ViewDefinitionModalProps, {}> {
+
+  public mounted: boolean;
```
Comment (piter75) [inline src/client/modals/view-definition-modal/view-definition-modal.tsx]: seems unused

File: src/client/components/hiluk-menu/hiluk-menu.tsx (hunk 1 above)
Comment (rafalnowak) [reply src/client/components/hiluk-menu/hiluk-menu.tsx]: in my opinion previous version was much less readable

**Pull request merged**